File: src/Piro.Domain/Enums/ServiceStatus.cs
namespace Piro.Domain.Enums;

/// <summary>Operational status of a service or check.</summary>
/// <remarks>Priority order (highest first): MAINTENANCE &gt; DOWN &gt; PARTIALLY_DOWN &gt; DEGRADED &gt; UP &gt; NO_DATA.</remarks>
public enum ServiceStatus
{
    NO_DATA,
    UP,
    DEGRADED,

    /// <summary>
    /// A multi-region check whose target is down in some, but not all, of the regions that measured it —
    /// a regional (partial) outage rather than a full one. Only ever produced by multi-region aggregation
    /// (<c>MultiRegionBatchTracker</c>); a single-region check is only ever UP or DOWN. More severe than
    /// DEGRADED (impaired but reachable), less severe than DOWN (down everywhere).
    /// </summary>
    PARTIALLY_DOWN,

    DOWN,
    MAINTENANCE,
    /// <summary>The check executor itself threw an unexpected exception. Not a service outage — internal error only.</summary>
    FAILURE
}

File: apps/admin/src/constants/serviceStatus.ts
export const SERVICE_STATUS = {
  NO_DATA:        "NO_DATA",
  UP:             "UP",
  DEGRADED:       "DEGRADED",
  PARTIALLY_DOWN: "PARTIALLY_DOWN",
  DOWN:           "DOWN",
  MAINTENANCE:    "MAINTENANCE",
  FAILURE:        "FAILURE",
} as const;

export type ServiceStatus = (typeof SERVICE_STATUS)[keyof typeof SERVICE_STATUS];

export const IMPACT_OPTIONS: { value: ServiceStatus; label: string }[] = [
  { value: "DEGRADED", label: "Degraded" },
  { value: "DOWN",     label: "Down" },
];

export const SERVICE_STATUS_LABEL: Record<ServiceStatus, string> = {
  NO_DATA:        "No data",
  UP:             "Up",
  DEGRADED:       "Degraded",
  PARTIALLY_DOWN: "Partially Down",
  DOWN:           "Down",
  MAINTENANCE:    "Maintenance",
  FAILURE:        "Check Error",
};
